pkg/resource/replication_group/post_set_output.go

```diff
@@ -42,9 +42,13 @@ func (rm *resourceManager) updateSpecFields(
 	latestCacheCluster, err := rm.describeCacheCluster(ctx, resource)
 	if err == nil && latestCacheCluster != nil {
 		setEngineVersion(latestCacheCluster, resource)
+		setMaintenanceWindow(latestCacheCluster, resource)
 	}
 }
 
+//TODO: for all the fields here, reevaluate if the latest observed state should always be populated,
+// even if the corresponding field was not specified in desired
+
```

Comment:
I think it should be fine to populate observed state fields here, the filterDelta() should determine if the diff is to be ignored.

Reply:
Agreed